docs/stock/data-grouping.md

```diff
@@ -1,24 +1,93 @@
 
-![data-grouping.png](data-grouping.png)
+```js
+Highcharts.stockChart('container', {
+  series: [{
+    dataGrouping: {
+      groupPixelWidth: 5
+    },
+    data: [...]
+  }]
+});
+```
 
-By default, the grouping [approximation](https://api.highcharts.com/highstock/plotOptions.series.dataGrouping.approximation) depends on the type of series:
 
+Data grouping in Highcharts Stock can be turned on or off using the enabled option. By default, it's set to true.
+
+```js
+Highcharts.stockChart('container', {
+    series: [{
+        dataGrouping: {
+            enabled: false
+        },
+        data: [...]
+    }]
+});
+```
+
+Here is an example of a difference between two series with the same data, but one of them is grouped:
+
+<iframe width="600" height="450" src="https://www.highcharts.com/samples/embed/stock/plotoptions/series-datagrouping-enabled" allowfullscreen></iframe>
+
+#### Overview of the most important data grouping options
+
+##### [Approximation](https://api.highcharts.com/highstock/plotOptions.series.dataGrouping.approximation)
+With the `approximation` option, you can decide how the final value for each group should be calculated. By default it depends on the series type: 
 *   Line type series use an _average_ approximation.
 *   Column type series compute the _sum_.
 *   Range type series compute the union's _range_.
 *   OHLC type series compute the union's open, high, low and close (OHLC) values.
 
-Grouping is activated when there are many data points in the chart. As well as increasing performance it makes it easier to spot trends in a chart.
 
-Data grouping is a Highcharts Stock feature and is enabled by default. To see dataGrouping options see the [API reference](https://api.highcharts.com/highstock/plotOptions.series.dataGrouping).
+but you can use the custom function to calculate the final value for each group. Here is an example of custom approximation function:
+
+<iframe width="600" height="600" src="https://www.highcharts.com/samples/embed/stock/plotoptions/series-datagrouping-approximation" allowfullscreen></iframe>
+
+ ##### [Units](https://api.highcharts.com/highstock/series.line.dataGrouping.units)
+The available data grouping units can be set using `units`. By default, it's an array of time units (millisecond, second, minute, hour, day, week, month, year) with their allowed multiples.
+
+```js
+Highcharts.stockChart('container', {
+    series: [{
+        dataGrouping:{
+            units: [['day',[1, 2, 3]]]
```

Comment:
Maybe add a comment explaining what the result of this will be? Like "This will allow only data to be grouped into one, two, or three days"